pkg/controller/cloneset/sync/cloneset_sync_utils.go

```diff
@@ -67,10 +69,10 @@ func (e expectationDiffs) isEmpty() bool {
 
 // This is the most important algorithm in cloneset-controller.
 // It calculates the pod numbers to scaling and updating for current CloneSet.
-func calculateDiffsWithExpectation(cs *appsv1alpha1.CloneSet, pods []*v1.Pod, currentRevision, updateRevision string) (res expectationDiffs) {
+func calculateDiffsWithExpectation(cs *appsv1alpha1.CloneSet, pods []*v1.Pod, currentRevision, updateRevision string, recorder record.EventRecorder) (res expectationDiffs) {
```

Comment:
Do not pass `recorder` into the calculation function. It should be recorded outside.